src/API/Router.ts

```diff
@@ -48,13 +48,22 @@ export class Router {
     }
 
     const getWorkSchema = {
+      query: {},
+    }
+
+    const getWorkByPublicKeySchema = {
       query: {
         publicKey: Joi.string().required(),
       },
     }
 
     this.koaRouter.get('/works/:id', RequestValidationMiddleware(getWorksSchema), this.getWork)
     this.koaRouter.get('/works', RequestValidationMiddleware(getWorkSchema), this.getWorks)
+    this.koaRouter.get(
```

Comment:
This is a problematic way of defining routes. What if other query params were valid for /works? How would you ensure the user puts 'publicKey' as the first param?

This logic should be part of the /works path as it is a means of filtering works. The filtering logic should be inside that route instead of making a new one. 

Further reading:
https://medium.com/@moschan/when-should-you-use-path-variable-and-query-parameter-a346790e8a6d